api/app/Actions/Auth/LoginAction.php

```diff
@@ -2,13 +2,28 @@
 
 namespace App\Actions\Auth;
 
+use App\Entity\User;
+use App\Repositories\User\UserRepository;
 use Illuminate\Auth\AuthenticationException;
 use Illuminate\Support\Facades\Auth;
 
 final class LoginAction
 {
+    private UserRepository $userRepository;
+
+    public function __construct(UserRepository $userRepository)
+    {
+        $this->userRepository = $userRepository;
+    }
+
     public function execute(LoginRequest $request): AuthenticationResponse
     {
+        $user = $this->userRepository->getByVerifiedEmail($request->getEmail());
+
+        if (!$user) {
+            throw new AuthenticationException('Email is not verified');
```

Comment:
You can create a separate exception `NotVerifiedEmailException` that is based on AuthenticationException.